### ISSUE TITLE
Decomposition issue in grocery.regions

### ISSUE THREAD

**Issue by spmaniato** [VerifiableRobotics/LTLMoP#22]
Cameron had to add the following hack to my specs in order to make the decomposition of "grocery.regions" work properly:

```
Group hack is between kitchen and table1, between entrance and table2
```

**Comment by cfinucane** [comment]
Modifying the spec to remove all "between A and B" phrases causes the decomposition to give strange results.  See attached picture and minimal breaking spec.

![broken decomposition](http://tomyumcorp.com/github_issues/grocery_decomp_broken.png)

http://tomyumcorp.com/github_issues/grocery_broken.spec

**Comment by jimjing** [comment]
fixed in f892ebff9457d0d175da79c2f23bd22d62e42d10
